umap/static/umap/js/modules/request.js

```diff
@@ -0,0 +1,160 @@
+// Uses `L._`` from Leaflet.i18n which we cannot import as a module yet
+import { Evented, DomUtil } from '../../vendors/leaflet/leaflet-src.esm.js'
+
+
+export class RequestError extends Error {}
+
+export class HTTPError extends RequestError {
+  constructor(message) {
+    super(message)
+    this.name = this.constructor.name
+  }
+}
+
+export class NOKError extends RequestError {
+  constructor(response) {
+    super(response.status)
+    this.response = response
+    this.status = response.status
+    this.name = this.constructor.name
+  }
+}
+
+const BaseRequest = Evented.extend({
```

Comment:
I'm still wondering if it's a good idea to extend the Leaflet logic here, let's talk one last time about that :)